### PR TITLE
Shannon 129 known good update [WIP]

### PR DESCRIPTION
**Pull request by shannon-lunarg**

### CODE REVIEW

File: layers/core_validation_types.h [hunk 1]
```diff
@@ -1697,6 +1697,7 @@ struct DeviceFeatures {
     VkPhysicalDeviceSeparateDepthStencilLayoutsFeaturesKHR separate_depth_stencil_layouts_features;
     VkPhysicalDevicePerformanceQueryFeaturesKHR performance_query_features;
     VkPhysicalDeviceTimelineSemaphoreFeaturesKHR timeline_semaphore_features;
+    VkPhysicalDeviceBufferDeviceAddressFeaturesEXT buffer_device_address_features;
```
Comment (Tony-LunarG) [inline layers/core_validation_types.h]: VkPhysicalDeviceBufferDeviceAddressFeaturesEXT and VkPhysicalDeviceBufferAddressFeaturesEXT are aliased and are the same.  No need to track them separately so I wouldn't add anything to DeviceFeatures or add extra checks for both structs.

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -4268,23 +4268,16 @@ bool CoreChecks::PreCallValidateCreateBuffer(VkDevice device, const VkBufferCrea
...
+            "VUID-VkBufferCreateInfo-flags-03338",
             "vkCreateBuffer(): the bufferDeviceAddressCaptureReplay device feature is disabled: Buffers cannot be created with "
             "the VK_BUFFER_CREATE_DEVICE_ADDRESS_CAPTURE_REPLAY_BIT_EXT set.");
     }
 
-    if ((pCreateInfo->usage & VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_EXT) &&
-        !enabled_features.buffer_address.bufferDeviceAddress) {
-        skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0,
-                        "VUID-VkBufferCreateInfo-usage-02606",
-                        "vkCreateBuffer(): the bufferDeviceAddress device feature is disabled: Buffers cannot be created with "
```
Comment (Tony-LunarG) [inline layers/buffer_validation.cpp]: Why is this check removed?

File: layers/state_tracker.cpp [hunk 3]
```diff
@@ -1133,6 +1133,12 @@ void ValidationStateTracker::PostCallRecordCreateDevice(VkPhysicalDevice gpu, co
         state_tracker->enabled_features.timeline_semaphore_features = *timeline_semaphore_features;
     }
 
+    const auto *buffer_device_address_features =
+        lvl_find_in_chain<VkPhysicalDeviceBufferDeviceAddressFeaturesEXT>(pCreateInfo->pNext);
+    if (buffer_device_address_features) {
```
Comment (Tony-LunarG) [inline layers/state_tracker.cpp]: No need to track both types since they are the same

File: layers/buffer_validation.cpp (hunk 2 above)
Comment (shannon-lunarg) [reply layers/buffer_validation.cpp]: `VUID-VkBufferCreateInfo-usage-02606` is the only VUID that was removed that I couldn't find a substitute for, so the check was removed.

Comment (Tony-LunarG) [reply layers/buffer_validation.cpp]: I see, newly legal to use that bit to create a buffer even if the feature isn't enabled

**Comment by shannon-lunarg** [comment]
@Tony-LunarG All checks are passing on GitHub, I'm still waiting for internal CI to start, but I believe I addressed all your previous comments.

**Pull request merged**